docs/collections/documents/tutorials/using_rest_api.md

```diff
@@ -399,8 +404,159 @@ print(resp.text)
+      self
+        .req(endpoint, {
+          body: { email, password },
+          method: "POST"
+        })
+        .then(({ jwt, ...data }) => {
+          self._headers.authorization = `bearer ${jwt}`;
+          resolve(data);
+        })
+        .catch(reject);
+    });
+  }
+
+  _handleResponse (response, resolve, reject) {
+    if (response.ok) {
+      resolve(response.json());
+    } else {
+      reject(response);
+    }
+  }
+
+  req (endpoint, { body, ...options } = {}) {
+    const self = this;
+    return new Promise(function (resolve, reject) {
+      fetch(self._url + endpoint, {
+        headers: self._headers,
+        body: body ? JSON.stringify(body) : undefined,
+        ...options
+      }).then((response) => self._handleResponse(response, resolve, reject));
+    });
+  }
+}
+const email = "nemo@nautilus.com";
+const password = "xxxxxx";
+const federationUrl = "https://api-gdn.paas.macrometa.io";
+
+const insertQuery = {
+  query:
+    "INSERT{'name' : 'Julie', 'company' : 'ABC', '_key' : 'Julie'} INTO testcollection"
+};
+const readQuery = {
+  query: "FOR doc IN testcollection RETURN doc"
+};
+const updateQuery = {
+  query:
+    "FOR c IN testcollection UPDATE c WITH{'company':'XYZ'} IN testcollection"
+};
+const upsertQuery = {
+  query:
+    "UPSERT {name: 'John'} INSERT {_key:'John', name: 'John', logins:1, updatedAt: DATE_NOW()} UPDATE {'logins': OLD.logins + 1, updatedAt: DATE_NOW()} IN testcollection"
+};
+
+const deleteDocumentsQuery = {
+  query: "FOR c IN testcollection REMOVE c IN testcollection"
+};
+const collectionName = "testcollection";
+
+const connection = new APIRequest(federationUrl);
+
+async function runningQueryFromCursor (query) {
+  await connection
+    .req("/_fabric/_system/_api/cursor", {
+      body: query,
+      method: "POST"
+    })
+    .then((document) => {
+      console.log("Cursor result:", document);
+      return document;
+    })
+    .catch((error) => console.log(error));
+}
+
+async function deleteCollection (collection) {
+  await connection
+    .req(`/_fabric/_system/_api/collection/${collection}`, {
+      method: "DELETE"
+    })
+    .then((document) => {
+      console.log("9. Delete collection " + collection);
+      return document;
+    })
+    .catch((error) => console.log(error));
+}
+
+async function createCollection (collection) {
+  await connection
+    .req("/_fabric/_system/_api/collection", {
+      body: { name: collection },
+      method: "POST"
+    })
+    .then((collection) => console.log("2. Collection created: ", collection))
+    .catch((error) => console.log(error));
+}
+
+const run = async function () {
+  /* -------------------- Log in (nemo@nautilus.com/xxxxxx) -------------------- */
+
+  await connection
+    .login(email, password)
+    .then(() => console.log("\n1. User authentication done!"))
+    .catch((error) => error);
+
+  /* -------------------------- Create document collection ------------------------- */
```

Comment:
Is there any reason not to put a blank line above each heading here? It would make it much easier to read.

Reply:
Not really, you can add a blank line